src/recurrent/rnn-time-step.js

```diff
@@ -0,0 +1,169 @@
+    outputs.push(output);
+    // 1+ indices
+    for (let i = 1, max = hiddenSizes.length; i < max; i++) {
+      output = this.getEquation(equation, output, equationConnection[i], hiddenLayers[i]);
+      outputs.push(output);
+    }
+
+    model.equationConnections.push(outputs);
+    equation.add(equation.multiply(model.outputConnector, output), model.output);
+    model.equations.push(equation);
+  }
+
+  /**
+   *
+   * @param {Number[]} input
+   * @returns {number}
+   */
+  runInput(input) {
+    this.runs++;
+    let model = this.model;
+    let errorSum = 0;
+    let equation;
+    while (model.equations.length < input.length - 1) {
+      this.bindEquation();
+    }
+    const outputs = [];
+
+    if (this.inputSize === 1) {
+      for (let inputIndex = 0, max = input.length - 1; inputIndex < max; inputIndex++) {
+        // start and end tokens are zeros
+        equation = model.equations[inputIndex];
+
+        const current = input[inputIndex];
+        const next = input[inputIndex + 1];
+        const output = equation.runInput([current]);
+        for (let i = 0; i < output.weights.length; i++) {
+          const error = output.weights[i] - next;
+          // set gradients into log probabilities
+          errorSum += Math.abs(error);
+
+          // write gradients into log probabilities
+          output.deltas[i] = error;
+          outputs.push(output.weights);
+        }
+      }
+    } else {
+      for (let inputIndex = 0, max = input.length - 1; inputIndex < max; inputIndex++) {
+        // start and end tokens are zeros
+        equation = model.equations[inputIndex];
+
+        const current = input[inputIndex];
+        const next = input[inputIndex + 1];
+        const output = equation.runInput(current);
+        for (let i = 0; i < output.weights.length; i++) {
+          const error = output.weights[i] - next[i];
+          // set gradients into log probabilities
+          errorSum += Math.abs(error);
+
+          // write gradients into log probabilities
+          output.deltas[i] = error;
+          outputs.push(output.weights);
+        }
+      }
+    }
+    //this.model.equations.length - 1;
+    this.totalCost = errorSum;
+    return errorSum;
+  }
+
+  runBackpropagate() {
+    for (let i = this.model.equations.length - 1; i > -1; i--) {
+      this.model.equations[i].runBackpropagate();
+    }
+  }
+
+
+  /**
+   *
+   * @param {Number[]|Number} [input]
+   * @param {Number} [maxPredictionLength]
+   * @param {Boolean} [isSampleI]
+   * @param {Number} temperature
+   * @returns {Number[]|Number}
+   */
+  run(input = [], maxPredictionLength = 1, isSampleI = false, temperature = 1) {
+    if (!this.isRunnable) return null;
+    const model = this.model;
+    while (model.equations.length < maxPredictionLength) {
+      this.bindEquation();
+    }
+    let lastOutput;
+    if (this.inputSize === 1) {
```

Comment:
Again we could combine these loops into 1 that has the logic in it for `if (this.InputSize === 1) {` to just contain the `runInput([input[i]]);` or `runInput(input[i]);`

Again Semicolons.